cypress/utils/CustomizedBridgeMultiChain.js

```
@@ -0,0 +1,207 @@
+import convertNumbersToNamed from "./convertNumbersToNamed";
+import capitalizeString from "./capitalizeString";
+import { providers } from "@ethersproject/experimental/node_modules/ethers";
+const PRIVATE_KEY_TEST_NEVER_USE =
+  "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+// address of the above key
+export const TEST_ADDRESS_NEVER_USE = new Wallet(PRIVATE_KEY_TEST_NEVER_USE)
+  .address;
+
+// ChainID: 1
+const MAINNET_CHAINID = "0x1";
+
+// ChainID: 69
+const OPTIMISM_KOVAN_CHAINID = "0x45";
+// ChainID: 10
+const OPTIMISM_MAINNET_CHAINID = "0xa";
+// ChainID: 421611
+const ARB_RINKEBY_CHAINID = "0x66eeb";
+
+class CustomizedBridgeMultiChain extends Eip1193Bridge {
+  constructor(defaultSigner, defaultProvider, signers, providers) {
+    super(defaultSigner, defaultProvider);
+
+    signers.forEach((signer, index) => {
+      ethers.utils.defineReadOnly(
+        this,
+        `signer${capitalizeString(convertNumberstoNamed(index + 1))}`,
+        signer
+      );
+    });
+
+    providers.forEach((provider, index) => {
+      ethers.utils.defineReadOnly(
+        this,
+        `provider${capitalizeString(convertNumberstoNamed(index + 1))}`,
+        provider || null
+      );
+    });
+
+    this.numProviders = providers.length;
+  }
+
+  send = async (...args) => {
+    console.debug("send called", ...args);
+    console.log("args -------", args);
+    const isCallbackForm =
+      typeof args[0] === "object" && typeof args[1] === "function";
+    let callback;
+    let method;
+    let params;
+    if (isCallbackForm) {
+      callback = args[1];
+      method = args[0].method;
+      params = args[0].params;
+    } else {
+      method = args[0];
+      params = args[1];
+    }
+
+    if (method === "wallet_switchEthereumChain") {
+      console.debug(
+        "send called with method wallet_switchEthereumChain",
+        ...args
+      );
+      console.log("args -------", args);
+
+      const chainId = args[1][0].chainId;
```

if you have maps like in the suggestion above, the rest of the code can just be
```
  assert(this.providers.has(chainid),'Unable to switch to chainId))
  assert(this.signers.has(chainid),'Unable to switch to chainId))
  this.provider = this.providers.get(chainId)
  this.signer = this.signer.get(chainId)
```

Also i would suggest throwing errors (if its possible) to signify you are unable to switch to a chain id, rather than default to a wrong one

---

Good suggestion.